File: 2023/12-01/main.js
const fs = require('fs');
let sum1 = 0;
let sum2 = 0;

fs.readFile('input.txt', 'utf-8', (error, string) => {
  string = string.split('\n');
  string.forEach((s) => {
    s = Number(s.replaceAll(/[^0-9]+/g, ''));
    if (s.toString().length == 1) {
      s = s * 10 + s;
    } else if (s.toString.length != 2) {
      s = Number(s.toString().split('').at(0) + s.toString().split('').at(-1));
    }
    sum1 += s;
  });
  console.log(`sum1 = ${sum1}`);

  string.forEach((s) => {
    string = s.split('\n');
    s = Number(
      s
        .replaceAll('one', 'o1e')
        .replaceAll('two', 't2o')
        .replaceAll('three', 't3e')
        .replaceAll('four', 'f4r')
        .replaceAll('five', 'f5e')
        .replaceAll('six', 's6x')
        .replaceAll('seven', 's7n')
        .replaceAll('eight', 'e8t')
        .replaceAll('nine', 'n9e')
        .replaceAll(/[^0-9]+/g, '')
    );

    if (s.toString().length == 1) {
      s = s * 10 + s;
    } else if (s.toString.length != 2) {
      s = Number(s.toString().split('').at(0) + s.toString().split('').at(-1));
    }

    sum2 += s;
  });
  console.log(`sum2 = ${sum2}`);
});

File: 2023/12-01/main.py
import re

sum1 = 0
sum2 = 0

with open("input.txt", "r") as f:
  strings = f.readlines()

  for string in strings:
    string = string.replace('\n', '')
    string1 = re.sub(r"[a-zA-Z]", "", string)
    string2 = string.replace('one', 'o1e').replace('two', 't2o').replace('three', 't3e').replace('four', 'f4r').replace('five', 'f5e').replace('six', 's6x').replace('seven', 's7n').replace('eight', 'e8t').replace('nine', 'n9e')
    string2 = re.sub(r"[a-zA-Z]", "", string2)

    if (len(string1) == 1):
      string1 = int(string1) * 10 + int(string1)
    elif (len(string1) != 2):
      string1 = int(string1[0]) * 10 + int(string1[-1])
    sum1 += int(string1)

    if (len(string2) == 1):
      string2 = int(string2) * 10 + int(string2)
    elif (len(string2) != 2):
      string2 = int(string2[0]) * 10 + int(string2[-1])
    sum2 += int(string2)
  print(f"sum1 = {sum1}")
  print(f"sum2 = {sum2}")

